Update Disk Check documentation for Windows (#10556)
DataDog/integrations-core: disk/README.md

The Disk check is enabled by default, and the Agent collects metrics on all local partitions. To configure the check with custom options, edit the `disk.d/conf.yaml` file, in the `conf.d/` folder at the root of your [Agent's configuration directory][2]. See the [sample disk.d/conf.yaml][3] for all available configuration options.

#### Note for Windows hosts
There are three scenarios where the Disk check can be used:

1. Monitoring physical drives

> Monitoring physical drives that are represented by a disk letter (e.g. C:\, D:\ etc...) is supported out of the box by the Disk check without any special considerations.

Reviewer: ```suggestion
  Monitoring physical drives that are represented by a disk letter (for example. C:\, D:\, etc.) is supported out of the box by the disk check without any special considerations.
```

2. Monitoring nested mount points

> Monitoring mounted folders within a filesystem requires Administrator permissions. That's because the underlying Windows function call [FindFirstVolumeMountPoint][9] requires administrative permissions.

Reviewer: ```suggestion
  Monitoring mounted folders within a filesystem requires Administrator permissions. This is because the underlying Windows function call [FindFirstVolumeMountPoint][9] requires administrative permissions.
```

> Monitoring mounted folders within a filesystem requires Administrator permissions. That's because the underlying Windows function call is [FindFirstVolumeMountPoint][9] which requires the caller to have administrative permissions.

Reviewer 2: ```suggestion
Monitoring mounted folders within a filesystem requires Administrator permissions. That's because the underlying Windows function call [FindFirstVolumeMountPoint][9] requires administrative permissions.
```
Nit: Made the sentence a bit shorter.

> To collect those metrics without granting Administrator permissions to the Agent, use the [PDH check][4] to collect mount point metrics from the corresponding perf counters.

Reviewer: ```suggestion
  To collect those metrics without granting Administrator permissions to the Agent, use the [PDH check][4] to collect mount point metrics from the corresponding perf counters.
```

3. Monitoring file shares

> Collecting mount point metrics for file shares on Windows is only supported by using the `create_mounts` option in the configuration.

Reviewer: ```suggestion
  Collecting mount point metrics for file shares on Windows is only supported by using the `create_mounts` option in the configuration.
```

> On Windows, each mounted folder is only visible to the user who mounted the share.

Reviewer: ```suggestion
  On Windows, each mounted folder is only visible to the user who mounted the share.
```

> The `create_mounts` option thus allows the Agent to create the mount points to monitor in the context of the Agent's user.

Reviewer: ```suggestion
  Therefore, the `create_mounts` option allows the Agent to create the mount points to monitor in the context of the Agent's user.
```
